examples/generative/corrdiff/training/training_loop.py

```diff
@@ -276,15 +274,22 @@ def training_loop(
                 loss = loss.sum().mul(loss_scaling / batch_gpu_total)
                 loss_accum += loss / num_accumulation_rounds
                 loss.backward()
-        wb.log({"training loss": loss_accum}, step=cur_nimg)
+
+        loss_sum = torch.tensor([loss_accum], device=device)
+        if dist.world_size > 1:
+            torch.distributed.all_reduce(loss_sum, op=torch.distributed.ReduceOp.SUM)
+        average_loss = loss_sum / dist.world_size
+        if dist.rank == 0:
+            wb.log({"training loss": average_loss}, step=cur_nimg)
```

Comment:
Probably better to use the stats_collector instead to ensure compatibility with the stats.jsonl and avoid having two ways of logging stats. The stats_collector already handles the broadcast and averaging across samples.


In my edm code I added a `log_to_tensorboard` function to `training_stats.Collector`

```python
    def log_to_tensorboard(self, writer: SummaryWriter):
        info = self.as_dict()
        try:
            nimg = info["Progress/kimg"]["mean"] * 1000
        except KeyError:
            nimg = None

        for k, v in info.items():
            for moment in v:
                writer.add_scalar(f"{k}/{moment}", v[moment], global_step=nimg)
```

You can easily modify this to log with wandb instead. And then in training_loop.py


```
train_collector = Collector()
for batch in batch:
      train_collector.report("Loss/loss", loss)
      
train_collector.update()
train_collector.log_to_wandb()
```

And similarly for the test loss.

Reply:
see https://github.com/NVIDIA/modulus/blob/ac0eff476656a389028637fb7358c3704cfb2f47/examples/generative/corrdiff/training/training_loop.py#L402

Reply:
you could event just put wandb logging at line 402

Reply:
The problem is, averaging in stats collector doesn't work as expected, and it essentially logs the rank0 loss. We can revisit that in a separate PR.